### DISCUSSION TITLE
Build arguments

### DISCUSSION THREAD

**Discussion by reichhartd** [remix-run/remix#6062] [Q&A] [answered]
We use Remix in combination with Cloudflare Pages. Within the loader and the actions we communicate via the GraphQL client `urql` with our backend.

Since we have set up several environments, the URL of our backends also changes. According to the [documentation](https://remix.run/docs/en/1.15.0/guides/envvars), you can adjust environment variables via Cloudflare, and then access them via the `context` within the loader and actions.
In our case, however, this would mean creating a new GraphQL client each time with the URL from the context, and then executing a query via that. Apart from the performance disadvantages of creating a new client each time, this also makes the code more complex.

Therefore my question: Is it possible to include build arguments during `remix build`? It would be sufficient for us to set the URL once during build.
The [documentation](https://remix.run/docs/en/1.15.0/guides/envvars#environment-variables) also mentions "build arguments". Unfortunately, it does not go into more detail.

An alternative solution I thought of is to create an `yml` or `json` file with the constants before the build. However, I would not prefer this solution.

**Comment by sergiodxa** [comment, marked as answer]
If you're using Cloudflare Pages, every request will run a new Worker invocation, which will create everything again, so there's no perf difference if you create the GQL client outside the loader/action or inside, because in both cases it will be a new client instance per request.

> Is it possible to include build arguments during remix build?

There's no build variables on Remix, you could write a proposal, right now only `process.env.NODE_ENV` is set at build time.
